src/Controller/AuthController.php

```diff
@@ -7,47 +7,43 @@
 
 namespace Lopi\Bundle\PusherBundle\Controller;
 
+use Lopi\Bundle\PusherBundle\Authenticator\ChannelAuthenticatorInterface;
 use Lopi\Bundle\PusherBundle\Authenticator\ChannelAuthenticatorPresenceInterface;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 
 /**
  * AuthController
  *
  * @author Richard Fullmer <richard.fullmer@opensoftdev.com>
  * @author Pierre-Louis Launay <laupi.frpar@gmail.com>
  */
-class AuthController extends Controller
+class AuthController
 {
+    private $pusherConfig;
+    private $authenticator;
+
+    public function __construnct(array $pusherConfig, ChannelAuthenticatorInterface $authenticator)
```

Comment:
Bad name ! :)

Reply:
woh, I just realized that use of configuration here will not work, because we can use `url` as config provider, so we will not have expected config values here. So I need to revert logic to use Pusher instance for parameters or make a new config service which will hold all configuration values in a correct way and use it in factory and other places. WDYT?